### DISCUSSION TITLE
Issue with OpenGL in anatomist

### DISCUSSION THREAD

**Discussion by manuegrx** [brainvisa/brainvisa.github.io#154] [Q&A] [answered]
Hi, 

It seems that Anatomist does not longer recognized my OpenGL version. 
The following message appears when I open Anatomist : `"Unrecognized OpenGL version"` and Anatomist crash when I want to open an image.
My OpenGL version is the following: 
```
glxinfo | grep "OpenGL version
OpenGL version string: 4.6.0 NVIDIA 470.199.02"
```
I downloaded a new casa with the latest developer image and I still have the same problem. 
I don't think that I changed anything recently regarding openGL in my computer. 

Has anyone encountered this type of problem or have any leads? 

Thank you, 

Emmanuelle

**Comment by denisri** [comment]
Hi,
I think I have never seen this error...
The string `"Unrecognized OpenGL version" ` does not exist in Anatomist source code. But I can find it in the Qt libs (`strings /usr/lib/x86_64-linux-gnu/libQt5Gui.so | fgrep "Unrecognized OpenGL version"` matches). So it's Qt which detects a problem with OpenGL at first.
Can you please output a bit more of `glxinfo` command please ? Like `server glx vendor` / `version`, `client glx vendor` / `version`, `GLX version`, `OpenGL vendor` / `renderer`... ?
Does `glxgears` work correctly ?
Have you recently performed updates on your host system ? Have you rebooted it after that ?
Denis

**Comment by manuegrx** [comment]
Thanks for your quick answer ! 
I don't think I performed updates on my host system recently (but  I haven't used anatomist for 2-3 months so I'm not sure when the problem started) 

As I am using casa to used Anatomist, should I check Qt and OpenGL on my host system or inside the container ? 
Could this be related to an issue on my host system and not in casa ? 
glxgears works on both (but with some "black inteference" in casa). 


In my host system: 
```
server glx vendor string: NVIDIA Corporation
server glx version string: 1.4
client glx vendor string: NVIDIA Corporation
client glx version string: 1.4
GLX version: 1.4
OpenGL vendor string: NVIDIA Corporation
OpenGL renderer string: Quadro K5000/PCIe/SSE2
OpenGL version string: 4.6.0 NVIDIA 470.199.02
```

In casa : 
```
server glx vendor string: Brian Paul
server glx version string: 1.4 Mesa 22.0.5
client glx vendor string: Brian Paul
client glx version string: 1.4 Mesa 22.0.5
GLX version: 1.4
OpenGL vendor string: Mesa/X.org
OpenGL renderer string: llvmpipe (LLVM 14.0.0, 256 bits)
OpenGL version string: 4.5 (Compatibility Profile) Mesa 22.0.5

```

**Comment by denisri** [comment, marked as answer]
Qt is used only inside the container, and works for other people. The problem is not in Qt I think, but Qt detects a problem in OpenGL and prints a warning about it, and he's right I think.
The problem is in OpenGL: to work properly with nvidia hardware and drivers, the drivers from the host linux system should be mounted and used from within the container. Here `glxinfo` says that the OpenGL/GLX implementation used inside the container is not nvidia but Mesa, so I guess the drivers are not correctly seen from within the container.
Can you try:
- exit the container
- restart is using the option `opengl=nv`
- try again `glxinfo` and `anatomist`. My guess is that it will work better ;)
- if yes, you can consider installing on the **host system** the package `nvidia-container-tools`: it contains tools helping to correcty detect and export nvidia drivers to the container when starting it. When installed, normally the option `opengl=nv` will not be beeded any longer, it will be done automatically.

See also https://brainvisa.info/casa-distro/technical.html#sing-opengl

**Comment by manuegrx** [comment]
It works correctly, thank you for your help !
